source/common/upstream/cluster_manager_impl.cc

```diff
@@ -130,6 +131,8 @@ ClusterManagerImpl::ClusterManagerImpl(const Json::Object& config, ClusterManage
       thread_local_slot_(tls.allocateSlot()), local_info_(local_info),
       cm_stats_(generateStats(stats)) {
 
+  config.validateSchema(Json::Schema::CLUSTER_SCHEMA);
```

Comment:
Cluster definition needs to be broken out so that it is validated in the createFromJson call. This is so that cluster definitions are checked when called loaded via the CDS API.